modin/experimental/backends/omnisci/query_compiler.py

```diff
@@ -86,6 +86,8 @@ class DFAlgQueryCompiler(BaseQueryCompiler):
     def __init__(self, frame, shape_hint=None):
         assert frame is not None
         self._modin_frame = frame
+        if len(self._modin_frame.columns) == 1 and shape_hint is None:
```

Comment:
```suggestion
        if shape_hint is None and len(self._modin_frame.columns) == 1:
```
this should work ever so slightly faster in the case when `shape_hint` is already known :)

Reply:
good point, changed as you suggested